code/model/modelrender.cpp

```diff
@@ -847,25 +847,33 @@ void model_render_add_lightning( model_draw_list *scene, model_render_params* in
 		switch ( smi->arc_type[i] ) {
 			// "normal", FreeSpace 1 style arcs
 		case MARC_TYPE_DAMAGED:
+			if ( Random::flip_coin() )	{
+				primary = Arc_color_damage_p1;
+			} else {
+				primary = Arc_color_damage_p2;
+			}
+
+			secondary = Arc_color_damage_s1;
+			break;
 		case MARC_TYPE_SHIP:
 			if ( Random::flip_coin() )	{
-				gr_init_color(&primary, std::get<0>(Arc_color_damage_p1), std::get<1>(Arc_color_damage_p1), std::get<2>(Arc_color_damage_p1));
+				primary = smi->arc_primary_color_1[i];
 			} else {
-				gr_init_color(&primary, std::get<0>(Arc_color_damage_p2), std::get<1>(Arc_color_damage_p2), std::get<2>(Arc_color_damage_p2));
+				primary = smi->arc_primary_color_2[i];
 			}
 
-			gr_init_color(&primary, std::get<0>(Arc_color_damage_s1), std::get<1>(Arc_color_damage_s1), std::get<2>(Arc_color_damage_s1));
+			secondary = smi->arc_secondary_color[i];
```

Comment:
Just for posterity, I suppose, it's unideal that this color info needs to be put on the submodel instance, even though it isn't *really* specific to the instance, but by this point things like the `ship_info` have been discarded awhile ago so that kind of data is no longer easily accessible. It's fine for now, but maybe something we should think about more in the future, if we have more 'ship_info-based rendering changes'.

Reply:
Yeah, that's precisely why I did it this way. At the very least passing the objnum down would help getting back to ship_info in cases like this where we might want access to it.